### PR TITLE
Fix `TestFleetManagedUpgrade` test + cleanup

### PR DESCRIPTION
**Pull request by ycombinator**
## What does this PR do?

This PR does a bit of cleanup to make the names of the three upgrade E2E tests - standalone, Fleet-managed, with download retries — be consistent.  It also consolidates all these tests into a single file, `upgrade_test.go`.

It also fixes the `TestFleetManagedUpgrade` test so it now passes.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~I have made corresponding changes to the documentation~
- [ ] ~I have made corresponding change to the default configuration files~
- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [ ] ~I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~
- [x] I have added an integration test or an E2E test

## How to test this PR locally

```
$ SNAPSHOT=true mage integration:single TestFleetManagedUpgrade
```
```
>>> Pulling latest ogc image
>>> Create SSH keys to use for SSH
>>> Creating zip archive of repo to send to remote hosts
>>> Creating ESS cloud 8.9.0-SNAPSHOT (at-shaunak-agent-testing-890-SNAPSHOT)
>>> Import layouts into ogc
>>> Bring up instances through ogc
>>> (linux/arm64/ubuntu/22.04[17e7]) Starting SSH connection to 35.238.82.143
>>> (linux/amd64/ubuntu/22.04[17e7]) Starting SSH connection to 104.198.213.70
>>> (linux/arm64/ubuntu/22.04[17e7]) Connected over SSH
>>> (linux/arm64/ubuntu/22.04[17e7]) Preparing instance
>>> (linux/arm64/ubuntu/22.04[17e7]) Running apt-get update
>>> (linux/amd64/ubuntu/22.04[17e7]) Connected over SSH
>>> (linux/amd64/ubuntu/22.04[17e7]) Preparing instance
>>> (linux/amd64/ubuntu/22.04[17e7]) Running apt-get update
>>> (linux/arm64/ubuntu/22.04[17e7]) Install build-essential and unzip
>>> (linux/amd64/ubuntu/22.04[17e7]) Install build-essential and unzip
>>> (linux/arm64/ubuntu/22.04[17e7]) Install golang 1.19.9 (arm64)
>>> (linux/arm64/ubuntu/22.04[17e7]) Copying repo
>>> (linux/arm64/ubuntu/22.04[17e7]) Running make mage and prepareOnRemote
>>> (linux/amd64/ubuntu/22.04[17e7]) Install golang 1.19.9 (amd64)
>>> (linux/amd64/ubuntu/22.04[17e7]) Copying repo
>>> (linux/amd64/ubuntu/22.04[17e7]) Running make mage and prepareOnRemote
>>> (linux/arm64/ubuntu/22.04[17e7]) Copying agent build elastic-agent-8.9.0-SNAPSHOT-linux-arm64.tar.gz
>>> (linux/arm64/ubuntu/22.04[17e7]) Waiting for stack to be ready
>>> (linux/arm64/ubuntu/22.04[17e7]) Will continue stack is ready
>>> (linux/arm64/ubuntu/22.04[17e7]) Starting sudo tests
>>> (linux/amd64/ubuntu/22.04[17e7]) Copying agent build elastic-agent-8.9.0-SNAPSHOT-linux-x86_64.tar.gz
>>> (linux/arm64/ubuntu/22.04[17e7]) Test output (sudo) (stderr): go: downloading github.com/rs/zerolog v1.27.0
>>> (linux/arm64/ubuntu/22.04[17e7]) Test output (sudo) (stdout): >> go test: remote-linux-arm64-ubuntu-2204-sudo.integration Testing
>>> (linux/arm64/ubuntu/22.04[17e7]) Test output (sudo) (stdout): exec: gotestsum --no-color -f standard-quiet --junitfile build/TEST-go-remote-linux-arm64-ubuntu-2204-sudo.integration.xml --jsonfile build/TEST-go-remote-linux-arm64-ubuntu-2204-sudo.integration.out.json -- -tags integration -test.run TestFleetManagedUpgrade github.com/elastic/elastic-agent/testing/integration
>>> (linux/amd64/ubuntu/22.04[17e7]) Waiting for stack to be ready
>>> (linux/amd64/ubuntu/22.04[17e7]) Will continue stack is ready
>>> (linux/amd64/ubuntu/22.04[17e7]) Starting sudo tests
>>> (linux/amd64/ubuntu/22.04[17e7]) Test output (sudo) (stderr): go: downloading github.com/rs/zerolog v1.27.0
>>> (linux/amd64/ubuntu/22.04[17e7]) Test output (sudo) (stdout): >> go test: remote-linux-amd64-ubuntu-2204-sudo.integration Testing
>>> (linux/amd64/ubuntu/22.04[17e7]) Test output (sudo) (stdout): exec: gotestsum --no-color -f standard-quiet --junitfile build/TEST-go-remote-linux-amd64-ubuntu-2204-sudo.integration.xml --jsonfile build/TEST-go-remote-linux-amd64-ubuntu-2204-sudo.integration.out.json -- -tags integration -test.run TestFleetManagedUpgrade github.com/elastic/elastic-agent/testing/integration
>>> (linux/arm64/ubuntu/22.04[17e7]) Test output (sudo) (stdout): ok  	github.com/elastic/elastic-agent/testing/integration	268.841s
>>> (linux/arm64/ubuntu/22.04[17e7]) Test output (sudo) (stdout): DONE 2 tests in 294.503s
>>> (linux/arm64/ubuntu/22.04[17e7]) Test output (sudo) (stdout): >> go test: remote-linux-arm64-ubuntu-2204-sudo.integration Test Passed
>>> (linux/amd64/ubuntu/22.04[17e7]) Test output (sudo) (stdout): ok  	github.com/elastic/elastic-agent/testing/integration	248.345s
>>> (linux/amd64/ubuntu/22.04[17e7]) Test output (sudo) (stdout): DONE 2 tests in 293.203s
>>> (linux/amd64/ubuntu/22.04[17e7]) Test output (sudo) (stdout): >> go test: remote-linux-amd64-ubuntu-2204-sudo.integration Test Passed
>>> Bring down instances through ogc
>>> Testing completed (4 successful)
>>> Console output written here: build/TEST-go-integration.out
>>> Console JSON output written here: build/TEST-go-integration.out.json
>>> JUnit XML written here: build/TEST-go-integration.xml
```

### CODE REVIEW

File: testing/integration/upgrade_test.go [hunk 1]
```diff
@@ -332,3 +326,216 @@ func (s *UpgradeStandaloneElasticAgent) TestUpgradeStandaloneElasticAgentToSnaps
...
+func (s *StandaloneUpgradeRetryDownloadTestSuite) getVersion(ctx context.Context) (*versionOutput, error) {
+	var currentVersion versionOutput
+	var err error
+
+	s.Eventually(func() bool {
+		args := []string{"version", "--yaml"}
+		var output []byte
+		output, err = s.agentFixture.Exec(ctx, args)
+
+		if err != nil {
```
Comment (AndersonQ) [inline testing/integration/upgrade_test.go:502]: [Go convention and consistency]
Please...
```suggestion
		output, err = s.agentFixture.Exec(ctx, args)
		if err != nil {
```

File: testing/integration/upgrade_test.go [hunk 2]
```diff
@@ -332,3 +326,216 @@ func (s *UpgradeStandaloneElasticAgent) TestUpgradeStandaloneElasticAgentToSnaps
...
+			s.T().Log(string(output))
+			return false
+		}
+
+		err = yaml.Unmarshal(output, &currentVersion)
+		if err != nil {
+			return false
+		}
+
+		return true
```
Comment (AndersonQ) [inline testing/integration/upgrade_test.go:512]: [Suggestion]
It can be simplified
```suggestion
		return err == nil
```

Comment (ycombinator) [reply testing/integration/upgrade_test.go:512]: Didn't mean to ignore this suggestion.  The PR was set to auto-merge so it got merged immediately upon seeing the approval.  Implemented this suggestion in https://github.com/elastic/elastic-agent/pull/2827.

File: testing/integration/upgrade_test.go (hunk 1 above)
Comment (ycombinator) [reply testing/integration/upgrade_test.go:502]: Didn't mean to ignore this suggestion.  The PR was set to auto-merge so it got merged immediately upon seeing the approval.  Implemented this suggestion in https://github.com/elastic/elastic-agent/pull/2827.

**Comment by pierrehilbert** [comment]
Tested on my side and works correctly!

**Pull request merged**